quixstreams/dataframe/windows/time_based.py

```diff
@@ -198,7 +270,10 @@ def process_window(
                 key, state, max_expired_window_start, collect
             )
 
-        return updated_windows, expired_windows
+        # Combine triggered windows with time-expired windows
+        all_expired_windows = triggered_windows + list(expired_windows)
```

Comment:
```suggestion
        all_expired_windows = itertools.chain(expired_windows, triggered_windows)
```

Two things here:
1. `itertools.chain()` allows us to not evaluate the iterator straight away. (`import itertools` is needed at the top)
2. It's better to process the expired windows first and only then triggered to keep the track of time in order downstream.